lib/semian.rb

```diff
@@ -178,7 +187,25 @@ def destroy(name)
   # the underlying semian resource (and sysV semaphore)
   def unregister(name)
     if resource = resources.delete(name)
-      resource.bulkhead.unregister_worker
+      resource.bulkhead.unregister_worker if resource.bulkhead
```

Comment:
@fw42 I noticed a failed test where it is possible for the bulkhead to be nil, I hadn't considered this before so I added this guard to handle that case.